robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot

```diff
@@ -1,50 +1,65 @@
     Set suite variable          &{program_engagement}
-    &{service} =                API Create Service              &{Program}[Id]
+    &{service} =                API Create Service              ${Program}[Id]
     Set suite variable          &{service}
+
     ${today} =                  Get Current Date                result_format=%Y-%m-%d
     Set suite variable          ${today}
     ${quantity} =               Generate Random String    2     [NUMBERS]
     Set suite variable          ${quantity}
 
 *** Test Cases ***
 Create a new service delivery using quick action
+     [Documentation]                This test loads the program engagement record, clicks on the new service delivery quick action and creates
+     ...                            new service delivery record.
+     [tags]                         W-037572  feature:Service Delivery
      Go To PMM App
-     Go To Page                  Details              ProgramEngagement__c                   object_id=&{program_engagement}[Id]
-     page should contain         &{program_engagement}[Name]
-     click quick action button   Create New Service Delivery
-     Load Page Object            NewServiceDelivery   ${ns}ServiceDelivery__c
-     verify current page title   Create New Service Delivery
-     populate modal form         Client=&{contact}[FirstName] &{contact}[LastName]
-     ...                         Service=&{service}[Name]
-     ...                         Quantity=${quantity}
-     Click Modal button          Save
+     Go To Page                     Details                        ProgramEngagement__c      object_id=${program_engagement}[Id]
+     Verify Details                 Program Engagement Name        contains                       ${contact}[FirstName] ${contact}[LastName] ${today}: ${program}[Name]
+     Click Quick Action Button      Create New Service Delivery
+     Verify Current Page Title      Create New Service Delivery
+     Populate Modal Form            Client=${contact}[FirstName] ${contact}[LastName]
+     ...                            Service=${service}[Name]
+     ...                            Quantity=${quantity}
+     Click Modal Button             Save
      Wait Until Modal Is Closed
-     verify page header          Program Engagement
-     Load Related List           Service Deliveries
-     click new related record link  &{contact}[FirstName] &{contact}[LastName] ${today}: &{service}[Name]
-     verify details             Service Delivery Name   contains                            &{contact}[FirstName] &{contact}[LastName] ${today}: &{service}[Name]
-     ${servicedelivery_id} =    Save Current Record ID For Deletion     ${ns}ServiceDelivery__c
-     ${service_id} =            Save Current Record ID For Deletion     ${ns}Service__c
-     ${programengagement_id} =  Save Current Record ID For Deletion     ${ns}ProgramEngagement__c
+     Verify Page Header             Program Engagement
```

Comment:
should we instead be using current page should be? can verification of page header could be added as part of that?

Reply:
Fixed, also removed 'Validate contact on quick action dialog' test as the contact is now autopopulated when the quick action button is clicked, so there is no need to validate contact on quickaction